vcomponent/src/vcomponent_hdmi_cec_device.h

```diff
@@ -0,0 +1,228 @@
+  LOGICAL_ADDRESS_PLAYBACKDEVICE3  = 11,
+  LOGICAL_ADDRESS_RESERVED1        = 12,
+  LOGICAL_ADDRESS_RESERVED2        = 13,
+  LOGICAL_ADDRESS_FREEUSE          = 14,
+  LOGICAL_ADDRESS_UNREGISTERED     = 15,
+  LOGICAL_ADDRESS_BROADCAST        = 15,
+  LOGICAL_ADDRESS_MAX              = 16
+
+} vCHdmiCec_logical_address_t;
+
+typedef enum
+{
+  CEC_VERSION_UNKNOWN = 0,
+  CEC_VERSION_1_2 = 1,
+  CEC_VERSION_1_2A = 2,
+  CEC_VERSION_1_3 = 3,
+  CEC_VERSION_1_3A = 4,
+  CEC_VERSION_1_4 = 5,
+  CEC_VERSION_2_0 = 6
+} vCHdmiCec_version_t;
+
+typedef enum
+{
+  PORT_TYPE_INPUT = 0,
+  PORT_TYPE_OUTPUT,
+  PORT_TYPE_UNKNOWN
+} vCHdmiCec_port_type_t;
+
+typedef enum
+{
+  POWER_STATUS_ON = 0,
+  POWER_STATUS_STANDBY,
+  POWER_STATUS_OFF,
+  POWER_STATUS_UNKNOWN
+} vCHdmiCec_power_status_t;
+
+typedef enum
+{
+  VENDOR_CODE_TOSHIBA = 0x000039,
+  VENDOR_CODE_SAMSUNG = 0x0000F0,
+  VENDOR_CODE_DENON = 0x0005CD,
+  VENDOR_CODE_MARANTZ = 0x000678,
+  VENDOR_CODE_LOEWE = 0x000982,
+  VENDOR_CODE_ONKYO = 0x0009B0,
+  VENDOR_CODE_MEDION = 0x000CB8,
+  VENDOR_CODE_TOSHIBA2 = 0x000CE7,
+  VENDOR_CODE_APPLE = 0x0010FA,
+  VENDOR_CODE_HARMAN_KARDON2 = 0x001950,
+  VENDOR_CODE_GOOGLE = 0x001A11,
+  VENDOR_CODE_AKAI = 0x0020C7,
+  VENDOR_CODE_AOC = 0x002467,
+  VENDOR_CODE_PANASONIC = 0x008045,
+  VENDOR_CODE_PHILIPS = 0x00903E,
+  VENDOR_CODE_DAEWOO = 0x009053,
+  VENDOR_CODE_YAMAHA = 0x00A0DE,
+  VENDOR_CODE_GRUNDIG = 0x00D0D5,
+  VENDOR_CODE_PIONEER = 0x00E036,
+  VENDOR_CODE_LG = 0x00E091,
+  VENDOR_CODE_SHARP = 0x08001F,
+  VENDOR_CODE_SONY = 0x080046,
+  VENDOR_CODE_BROADCOM = 0x18C086,
+  VENDOR_CODE_SHARP2 = 0x534850,
+  VENDOR_CODE_VIZIO = 0x6B746D,
+  VENDOR_CODE_BENQ = 0x8065E9,
+  VENDOR_CODE_HARMAN_KARDON = 0x9C645E,
+  VENDOR_CODE_UNKNOWN = 0
+} vCHdmiCec_vendor_code_t;
+
+typedef struct {
+    bool allocated[LOGICAL_ADDRESS_MAX];
+} vCHdmiCec_logical_address_pool_t;
+
+struct vCHdmiCec_device_info_t
+{
+  /*Variables to manage a non-binary tree of devices*/
+  unsigned int number_children;
+  struct vCHdmiCec_device_info_t* parent;
+  struct vCHdmiCec_device_info_t* first_child;
+  struct vCHdmiCec_device_info_t* next_sibling;
+
+  /*Device Information*/
+  vCHdmiCec_device_type_t type;
+  vCHdmiCec_version_t version;
+  unsigned int physical_address;
+  unsigned int logical_address;
+  bool active_source;
+  unsigned int vendor_id;
+  unsigned short parent_port_id;
+  vCHdmiCec_power_status_t power_status;
+  char osd_name[MAX_OSD_NAME_LENGTH];
+} vCHdmiCec_device_info_t;
+
+typedef struct
+{
+  char* str;
+  int val;
+} strVal_t;
+
+/**
+* Creates a map of devices in a parent child n-ary tree.
+* Recursively loads the device map from passed in profile instance
+* To load from root device, prefix shall be
+*   "hdmicec/device_map/0"
+*
+*  Returns pointer to newly created vCHdmiCec_device_info_t
+*
+*/
+struct vCHdmiCec_device_info_t* vCHdmiCec_Device_CreateMapFromProfile (ut_kvp_instance_t* instance, char* profile_prefix);
```

Comment:
you'll need to fill out all your doxygen commnets, try chatGPT to do that for you.